src/analysis/sccp.rs

```diff
@@ -179,10 +179,12 @@ where
                 self.cfg_worklist.push_back(false_branch);
             }
             LatticeValue::Top => {
-                // TODO: Not really sure what to do here.
+                radeco_warn!("`cond_val` is undefined");
+                self.cfg_worklist.push_back(true_branch);
+                self.cfg_worklist.push_back(false_branch);
             }
             LatticeValue::Const(cval) => {
-                if cval == 1 {
+                if cval == 0 {
```

Comment:
why the reverse here?

Reply:
When I tried the code like
```c
// do something1
if 8 {
    // do something2
} else {
    return;
}
```
With `if cval != 0` instead of `if cval == 1`, the result became

```
// do something1
return;
```


So I reveresed the condition.
Actually I am not sure why this way goes well.

Reply:
@sushant94 do you know why this might happen?

Reply:
@kriw Can you post the asm/esil code for above? That may tell us something as to why this is happening. It might be because that constant is a comparison for `zf`, i.e., is zf == 0, which happens if the condition is true (and hence the result that you observe above).